### ISSUE TITLE
Set a higher quota for app backups

### ISSUE THREAD

**Issue by John-Rodriguez-Git** [stevesoltys/backup#822]
Im a heavy cromite user and a tab junkie/messie and have over 3GB User Data. I backup to my 512GB nvme and would want to set a higher quota.
Why was 
https://github.com/seedvault-app/seedvault/pull/750/commits/03d2946c93813602c0efd0d7d80b9d0a5852f367
done?

Linking to https://github.com/seedvault-app/seedvault/issues/183 and https://github.com/seedvault-app/seedvault/issues/231 as the described solution to increase the quota is not available anymore since that commit.

**Comment by grote** [comment]
I don't think we should bring back the option to manually set the quota. Instead, we could bump the quota to a higher value, 5 GB maybe.

**Comment by Waffle42** [comment]
Please bring it back.

**Comment by oxalica** [comment]
I updated my phone system from LineageOS 21.0 to 22.1 (major update) yesterday and did a storage reset for clean installation. It turns out some of my apps failed to restore due to "quote exceeded".
The most painful one is [fcitx5-android](https://github.com/fcitx5-android/fcitx5-android), the Chinese input method which contains my own manually-trained dictionary. It is 53MB in the backup but failed to restore due to quota issue. I guess I turned on "unlimited quota" on backup, but now that option is gone from expert mode. How can I restore my own existing data now?

**Comment by grote** [comment]
This ticket is unrelated to restore. If restore failed due to quota exceeded, then you did the backup with an old version and limited quota. The current quota of the recent version can handle 53MB just fine.

**Comment by Waffle42** [comment]
> I don't think we should bring back the option to manually set the quota. Instead, we could bump the quota to a higher value, 5 GB maybe.

Is there a reason for why you wouldn't want to bring it back? Shouldn't the power be held by the user?

**Comment by oxalica** [comment]
> This ticket is unrelated to restore. If restore failed due to quota exceeded, then you did the backup with an old version and limited quota. The current quota of the recent version can handle 53MB just fine.

Okay maybe I misunderstood the issue. I'm on seedvault 15-5.2 which should contain the 1GB default change mentioned in the issue description. But in "Restore backup" I click the 2 days ago backup, and Fcitx5 shows "Last backup: December 10, 2024 (52MB)". When I select it and click "Restore backup", then it shows "Quota exceeded". Do you mean the first information is incorrect and the app is not actually backed up to my remote storage? This is also misleading to me...

**Comment by grote** [comment]
> But in "Restore backup" I click the 2 days ago backup, and Fcitx5 shows "Last backup: December 10, 2024 (52MB)". When I select it and click "Restore backup", then it shows "Quota exceeded". 

So the backup you did on December 10 most likely had a quota limit of 50 MB. Please do a new backup of this app with 5.2 and if that doesn't work for some reason, please open a new ticket with a log.

> Is there a reason for why you wouldn't want to bring it back? Shouldn't the power be held by the user?

The UX philosophy of Seedvault is not to introduce an endless number of little toggles to modify its behavior in all details, but to function in a way that works well for the majority of users. Also see https://github.com/seedvault-app/seedvault/issues/822#issuecomment-2546451919

**Comment by tio-trom** [comment]
Would be great if we can chose the maximum quota for an app. I have apps that are over 3GB in size, like offline maps and such, and I have plenty of storage to back them up. Now I cannot...

**Comment by grote** [comment]
Offline maps can be downloaded again easily, they should really not be part of backup. Ideally, the app excludes such things from all backups.

**Comment by tio-trom** [comment]
> Offline maps can be downloaded again easily, they should really not be part of backup. Ideally, the app excludes such things from all backups.

I agree, but right now it is not possible to exclude these from backing up no?

**Comment by grote** [comment]
You can exclude apps in the backup status screen, overflow menu top right.

**Comment by tio-trom** [comment]
Yes but I cannot exclude the app data. I still want to backup my navigation apps...but currently I cannot since the offline maps are seen as part of the app and they exceed 1GB....

**Comment by grote** [comment]
I'd suggest reporting the issue to the app developers.

**Comment by tio-trom** [comment]
> I'd suggest reporting the issue to the app developers.

Arent all app being dealt with the same way by Seedvault? Meaning Seedvault wants to backup the apps plus data for each app?

**Comment by grote** [comment]
Seedvault uses the official Android backup API, so apps are responsible for what data they deliver to backup. Data that can always be downloaded again (and on top of this is huge) should be excluded from backup.

**Comment by John-Rodriguez-Git** [comment]
Would you accept patches to make the backup quota configurable in the expert settings?
Or can you put the quota to 5GB as you suggested in your first comment?
https://github.com/seedvault-app/seedvault/blame/android16/app/src/main/java/com/stevesoltys/seedvault/settings/SettingsManager.kt#L256

My Cromite UserData has grown to over 3GiB.

**Comment by MrRinkana** [comment]
> Data that can always be downloaded again (and on top of this is huge) should be excluded from backup.

But this does not make sense to me, the apps themselves can also be downloaded again - yet backing up without them would make restoring a pita (having to manually install every app again).

Same goes for large appdata - i can download that much faster from my nextcloud or from an sd card for that matter than from random map apps, spotify or w/e that's blocked by this.. Restoring all that can be a fair bit of manual work aswell..

I understand the default being a compromise between the time the backup takes, writes done and convience/completeness but why limit the choice?

Maybe I am using an app where the download servers have closed? Or it's actually non redownloadable data?

Maybe the most elegant solution is to allow users to manually override the limit *per app*, by clicking on the "not backed up, exceeded quota" notice rather than globally?

**Comment by grote** [comment]
> Would you accept patches to make the backup quota configurable in the expert settings?
> Or can you put the quota to 5GB as you suggested in your first comment?

You can submit an PR to set the limit to 5 GB.

**Comment by Eridyne** [comment]
The current hard limit is, I think, a very inappropriately opinionated approach.

A backup/restore system's sole purpose is to safely store data selected for backup, then correctly put it back when requested. It can include supporting functions, obviously - but that's it. Do backup, do restore.

If an app presents gigs of maps for backup in an attempt to save some users' limited cellular data, or a user has gigs of emails in their mail client or gigs of browser profile... okay. Back it up.

Warn on large backups? Sure. Dislike large backups? You do you.

But a backup system should never decree what, or how much, the user is permitted to back up, especially based on an opinion of how backups ought to be. That's out of scope, not its job in any way.

Technical issues would be different, but that's not the case here.
